SMLHelper/Assets/CustomPrefab.cs

```diff
@@ -0,0 +1,119 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using SMLHelper.Assets.Gadgets;
+using SMLHelper.Assets.PrefabTemplates;
+using SMLHelper.Handlers;
+using UnityEngine;
+
+namespace SMLHelper.Assets;
+
+public delegate IEnumerator PrefabFactoryAsync(TaskResult<GameObject> gameObject);
+
+/// <summary>
+/// Specifies the contract for a custom prefab.
+/// </summary>
+public interface ICustomPrefab
+{
+    /// <summary>
+    /// The prefab info for this custom prefab.
+    /// </summary>
+    PrefabInfo Info { get; }
+    
+    /// <summary>
+    /// Function which constructs a game object as this prefab.
+    /// </summary>
+    PrefabFactoryAsync Prefab { get; }
+    
```

Comment:
Optional sync version has been added.

Optional PostProcess is no longer necessary as you can modify all the fields a `PostProcess` uses on the PrefabInfo property.

Similar to the suggested EnhancePrefab method is already available. Since `ICustomPrefab` has a `SetPrefab` overload that accepts a `Func<IOut<GameObject>, IEnumerator>`, you can pass a method of signature `IEnumerator(IOut<GameObject>)` to create your own EnhancePrefab method.
You're allowed to use any template as well as they all have a `GetPrefabAsync` to get your processed template.

Reply:
> Optional PostProcess is no longer necessary as you can modify all the fields a PostProcess uses on the PrefabInfo property.

It is if you do not want to run smlhelpers processor at all or to do it yourself.

> Similar to the suggested EnhancePrefab method is already available. Since ICustomPrefab has a SetPrefab overload that accepts a Func<IOut<GameObject>, IEnumerator>, you can pass a method of signature IEnumerator(IOut<GameObject>) to create your own EnhancePrefab method.
You're allowed to use any template as well as they all have a GetPrefabAsync to get your processed template.

I want there to be a Non coroutine option for processing after smlhelpers processor is complete please and thank you.